lib/improver/tests/ensemble_calibration/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py

```diff
@@ -292,136 +305,161 @@ def test_basic_mean_predictor_bad_value(self):
-        cube = set_up_temperature_cube()
-
-        forecast_predictor = cube.collapsed("realization", iris.analysis.MEAN)
-        forecast_variance = cube.collapsed(
-            "realization", iris.analysis.VARIANCE)
-        truth = cube.collapsed("realization", iris.analysis.MAX)
-
         predictor_of_mean_flag = "mean"
-
-        plugin = Plugin()
         distribution = "gaussian"
+        plugin = Plugin()
         result = plugin.crps_minimiser_wrapper(
-            initial_guess, forecast_predictor, truth, forecast_variance,
+            self.initial_guess_for_mean,
+            self.temperature_forecast_predictor_mean,
+            self.temperature_truth_cube,  self.temperature_forecast_variance,
             predictor_of_mean_flag, distribution)
         self.assertIsInstance(result, np.ndarray)
         self.assertEqual(result.dtype, np.float32)
         self.assertArrayAlmostEqual(
-            result, [-0.059093, -0.099905, 0.008257, 1.009563])
+            result, self.expected_mean_coefficients)
 
     @ManageWarnings(
         ignored_messages=["Collapsing a non-contiguous coordinate.",
-                          "Minimisation did not result in convergence"])
-    def test_basic_normal_realizations_predictor(self):
+                          "Minimisation did not result in convergence",
+                          "divide by zero encountered in",
+                          "invalid value encountered in"],
+        warning_types=[UserWarning, UserWarning, RuntimeWarning,
+                       RuntimeWarning])
+    def test_basic_realizations_predictor(self):
         """
         Test that the plugin returns a numpy array.
         The ensemble realizations are the predictor.
         """
-        initial_guess = [5, 1, 0, 1, 1, 1]
-        initial_guess = np.array(initial_guess, dtype=np.float32)
-        cube = set_up_temperature_cube()
-
-        forecast_predictor = cube.copy()
-        forecast_variance = cube.collapsed(
-            "realization", iris.analysis.VARIANCE)
-        truth = cube.collapsed("realization", iris.analysis.MAX)
-
         predictor_of_mean_flag = "realizations"
-
-        plugin = Plugin()
         distribution = "gaussian"
+        plugin = Plugin(max_iterations=5)
         result = plugin.crps_minimiser_wrapper(
-            initial_guess, forecast_predictor, truth, forecast_variance,
+            self.initial_guess_for_realization,
+            self.temperature_forecast_predictor_realizations,
+            self.temperature_truth_cube,  self.temperature_forecast_variance,
             predictor_of_mean_flag, distribution)
         self.assertIsInstance(result, np.ndarray)
         self.assertEqual(result.dtype, np.float32)
         self.assertArrayAlmostEqual(
-            result, [6.24021609e+00, 1.35694934e+00, 1.84642787e-03,
-                     5.55444682e-01, 5.04367388e-01, 6.68575194e-01])
+            result, self.expected_realizations_coefficients,
+            decimal=4)
```

Comment:
Maybe we want to just set this `decimal=` value as a top level constant and use it in all the `assertArrayAlmostEqual` methods? Another approach would be that we'll allow differences so long as they're less than <0.5% (or some other suitable value) of the original data? (thinking out loud here...)

Reply:
Do you mean to set a lower precision requirement for `assertArrayAlmostEqual`, specifically for the ensemble calibration unit tests? At the moment, the default is 6 decimal places as [seen in Iris](https://github.com/SciTools/iris/blob/b20964241d4523bee6a91186a55e6988363c0a9a/lib/iris/tests/__init__.py#L618). At the moment, I've only amended the `decimal` option where I've seen a bit of variability between the machines that I've tested on. Do you think that it would be appropriate to use a lower precision e.g. 4 decimal places for all ensemble calibration unit tests?

Reply:
Hmm, I guess it depends on how stable the algorithm is once you move to float64 - maybe 6 decimal places will be okay then, and we won't have to change it. Given that you said that, even when the coefficients varied quite significantly (when using float32) across different platforms, that it didn't affect the overall correction/results, I suspect there's a case for using a reduced precision, like e.g: 4 decimal places. It really does relate to the `tol` argument used for the optimization though!

Reply:
As mentioned elsewhere and discussed, I've updated the tests on the coefficients to be to 4 decimal places by default for these ensemble calibration tests.